lib/google_drive.rb

```diff
@@ -122,34 +113,24 @@ def self.saved_session(path = nil, proxy = nil, client_id = nil, client_secret =
           :application_version => "0.4.0"
       )
       auth = client.authorization
-      auth.client_id = client_id
-      auth.client_secret = client_secret
-      auth.scope = [
-          "https://www.googleapis.com/auth/drive",
-          "https://spreadsheets.google.com/feeds/"
-      ]
-      auth.redirect_uri = "urn:ietf:wg:oauth:2.0:oob"
-
-      if token_data
+      auth.client_id = config.client_id
+      auth.client_secret = config.client_secret
+      auth.scope = config.scope.join(' ')
+      auth.redirect_uri = config.redirect_uris.first
```

Comment:
Do you know of any use case where you want to specify something other than "urn:ietf:wg:oauth:2.0:oob" here? Otherwise can you just hard-code "urn:ietf:wg:oauth:2.0:oob" as in the original code?

The command line authentication in the else-clause at line 124 only works with "urn:ietf:wg:oauth:2.0:oob". If you directly specify refresh_token in the config, then you don't need to provide redirect_uri here.